### DISCUSSION TITLE
Why use cgo but not pure go for indexer?

### DISCUSSION THREAD

**Discussion by ayanamist** [awslabs/soci-snapshotter#63] [Q&A]
I notice that it's using a [C module](/awslabs/soci-snapshotter/tree/main/c) to generate indices.
Why not use pure go like standard library is doing?
This design also causes issues like #60

**Comment by rdpsin** [comment]
We use [`zlib`](https://github.com/madler/zlib/) to generate the dictionary data as well as decompress spans using that data. To facilitate that, we make use of certain C functions such as `inflateInit2`, `inflateSetDictionary` and `inflatePrime` ([Link](https://www.zlib.net/manual.html)). AFAIK, there is no Go package that provides this raw functionality. 

In the fullness of time, we want to migrate the C code to Go, using some kind of Go abstraction layer over the C zlib API. Once we do that, it may be possible to use this library as a Go module.
